Feat/component activities/page render cards activities

andreajhl opened the pull request:
Show list of activities in the route '/ ativities'

-render the title component to show the title of the page
-Pagination: add component to make a pagination of the components from the front (shows 9 cards at the same time), make a get request to the api to bring all the activities to render, render two buttons to move the cards back or forward show
-Activities Cards: receives the data from the page (only the ones that must be shown on that page) to map them and pass the data of each object to each card
-activity card: receives by props the data to show, image, title, and description
-activitiesList: renders the two main components, title for the title and pagination that contains all the logic of the card rendering
-you don't have defined styles yet, these are just sample

postData: over the weekend I will style the forms and this component

![1](https://user-images.githubusercontent.com/75461111/140187721-ff435ac2-2a7a-43fb-a936-1b04119ceae1.png)

Commit:
feat: :sparkles: add route "/activities" for render activities list

Commit:
feat: :sparkles: add component "title"and "Pagination"

Commit:
feat: :sparkles: step through the component array to pass the data to each card

Commit:
feat: :sparkles: contiene logica para agregar un paginado desde el front, asi se muestra solo 9 actividades por pantalla,

This sets the items to be displayed per page in a local state and sends them to activities cards for them to go through.

Commit:
feat: :sparkles: add components to render card, show image title and description

Commit:
feat: add function to remove the "<p> </p>" and "<strong> </strong>" by a space

Commit:
add request http get list activities

andreajhl committed:
Merge branch 'development' into feat/component-activities/page-render-cards-activities

saralavanchy commented on src/Components/Activities/ActivitiesCards/ActivitiesCards.js:
```diff
@@ -0,0 +1,11 @@
+
+    return (
+        <div>
+            {activities ? activities.map(e=><ActivityCard props={e} key={e.id}/>) : <p>No hay Actividades</p>}
```
This is not a blocker, but for future components texts like "No hay Actividades" can be on a constants file so they can be reused or replaced easily.

saralavanchy replied:
instead of creating a new component it can be a function on Activities Pagination

saralavanchy commented on src/Components/Activities/ActivitiesCards/ActivitiesCards.js:
```diff
@@ -0,0 +1,11 @@
+import ActivityCard from '../ActivityCard/ActivityCard';
+const ActivitiesCards = ({activities}) => {
```
This component is not necessary, its adding complexity to a simple task.

saralavanchy commented on src/Components/Activities/ActivitiesList.js:
```diff
@@ -1,32 +1,15 @@
 import React from 'react';
+import Title from '../Title/Title';
+import PaginationActivities from './ActivitiesPagination/ActivitiesPagination';
```
the name of the component is ActivitiesPagination  and not PaginationActivities

saralavanchy commented on src/Components/Activities/ActivitiesPagination/ActivitiesPagination.js:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    const next=()=>{
+        const totalElementos= activities.length
```
if your are escribiendo all the code in english it should be totalOfItems

saralavanchy commented on src/Components/Activities/ActivitiesPagination/ActivitiesPagination.js:
```diff
@@ -0,0 +1,56 @@
+import {getAllActivities} from '../../../Services/activityService';
+const cards = 9;
+
+const ActivitiesPagination = () => {
```
you can make a Generic Pagination component passing by attribute the items list and then you can have a list of currentItems instead of just items and also the  <ActivitiesCards activities={items} /> can be send by attribute on a function :
```
const ListPagination = ({items, showItemsListComponent}) => {
    const [currentItems, setCurrentItems]=useState(null);
    const [currentPag, setCurrentPag]= useState(0);
    
   useEffect(()=>{
        if(items.length>1)setCurrentItems([...items].splice(0,cards))
    },[items])

....

return (
        <div className='paginacion'>
            <div className='paginacion_activities'>
                {showItemsListComponent(currentItems)}
            </div>
          <div className='paginacion_buttons'>
                <button className='paginacion_btn' onClick={()=>prev()}><ArrowLeft/></button>
                <button className='paginacion_btn' onClick={()=>next()}><ArrowRight/></button>
          </div>
        </div>
    )
}
```

saralavanchy commented on src/Components/Activities/ActivityCard/ActivityCard.js:
```diff
@@ -0,0 +1,13 @@
+import {replaceCKEditorText} from '../../common/ckEditor/setCKEditorText'
+const ActivityCard = ({props}) => {
+    const {name,description,image}=props
```
props on props:
```
const ActivityCard = ({name,description,image}) => {
```

saralavanchy commented on src/Components/Activities/ActivitiesList.js:
```diff
@@ -1,32 +1,15 @@
+import PaginationActivities from './ActivitiesPagination/ActivitiesPagination';
 import '../CardListStyles.css';
-
 const ActivitiesList = () => {
```
if you make the ActivitiesPagination component generic you should get the activities from the service here and create a function to return the cards:
```
const showItemsListComponent = (items) => (<ActivitiesCards activities={items} />);
```

saralavanchy commented on src/Components/Activities/ActivitiesList.js:
```diff
@@ -1,32 +1,15 @@
-                }
-            </ul>
+            <Title title={'Actividades'} id={'activitiesList'}/>
+            <PaginationActivities/>
```
if you maje ActivitiesPagination generic this should be something like 
```
<ListPagination items={activities} showItemsListComponent={showItemsListComponent } />
```

saralavanchy commented on src/Components/common/ckEditor/setCKEditorText.js:
```diff
@@ -3,3 +3,6 @@ export const setCKEditorText = (values,field)=>{
     return values
 };
 
+export const replaceCKEditorText = (text)=>{
```
I think that alread exists a function to do that on the project

Commit:
feat: :fire: remove deprecated code

Commit:
add text const folder

Commit:
feat: :sparkles: add component to render cards

use the ckeditor function to remove the html code from the description

Commit:
feat: :sparkles: add reusable component to create pagination from front end

receives a function that renders the list and the items in an array that will divide

Commit:
add text const folder

Commit:
modify the above function to also remove "<strong> </strong>"

Commit:
feat: :sparkles: add component to render cards

use the ckeditor function to remove the html code from the description and use constants for texts, so they can be changed more easily in the future

Commit:
add component that renders the pagination with the cards and the title of the page

andreajhl committed:
Merge branch 'development' into feat/component-activities/page-render-cards-activities
